notebook/gateway/handlers.py

```diff
@@ -216,9 +252,11 @@ def on_message(self, message):
             self._write_message(message)
 
     def _write_message(self, message):
-        """Send message to gateway server."""
+        """
+        Send message to gateway server.
+        """
         try:
-            if not self.disconnected and self.ws is not None:
+            if not self.disconnected and self.ws:
```

Comment:
was curious. Aren't the two complimentary? As in, if there is no ws, there can never be a connection, right? And even if there is a ws, can it be disconnected? Why do we need `self.ws` in the first place. the value of `self.ws and not self.disconnected` gives us the state of notebook being connected, and vice versa

Reply:
This is not something I have wrote and I don't want to mess with the conditions just made these conditions in a more pythonic way.